src/main/scala/edu/ie3/mobsim/MobilitySimulator.scala

```diff
@@ -402,23 +405,23 @@ final class MobilitySimulator(
     *   Maximum permissible distance between point of interest and charging
     *   station
     * @return
-    *   An [[Future]] onto an [[Option]] of the target charging stations' UUID
-    *   as well as information about the movement
+    *   An [[Option]] of the movement towards a charging station, if one was
+    *   made
     */
   private def handleArrivingEv(
       ev: ElectricVehicle,
       pricesAtChargingStation: Map[UUID, Double],
       availableChargingPoints: Map[UUID, Int],
       maxDistance: ComparableQuantity[Length]
-  ): Future[Option[(UUID, Movement)]] = Future {
+  ): Option[Movement] =
     chooseChargingStation(
       ev,
       pricesAtChargingStation,
       availableChargingPoints,
       MobilitySimulator.seed,
       maxDistance
     ).map { cs =>
-      val availableChargingPointsAtStation: Int =
+      val availableChargingPointsAtStation =
         availableChargingPoints.getOrElse(cs, 0)
```

Comment:
So i learned about Design by contract recently and read advice about checking Preconditions of specific methods. In this case this would mean adding a `require(availableChargingPointsAtStation >= 0)` which seems pretty sensible in this case

Reply:
I definitely see the point if we'd call another method here that could check on `availableChargingPointsAtStation` again. Checking the variable right after the if condition that requires the same thing not so much, tbh.